internal/controller/management/projects/projects.go

```diff
@@ -341,6 +364,194 @@ func (r *reconciler) ensureProjectAdminPermissions(
 	return nil
 }
 
+func (r *reconciler) ensureDefaultProjectRoles(
+	ctx context.Context,
+	project *kargoapi.Project,
+) error {
+	logger := logging.LoggerFromContext(ctx).WithFields(log.Fields{
+		"project":   project.Name,
+		"name":      project.Name,
+		"namespace": project.Name,
+	})
+
+	const adminRoleName = "admin"
+	const viewerRoleName = "viewer"
+	allRoles := []string{adminRoleName, viewerRoleName}
+
+	for _, saName := range allRoles {
+		saLogger := logger.WithField("serviceAccount", saName)
+		if err := r.createServiceAccountFn(
+			ctx,
+			&corev1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      saName,
+					Namespace: project.Name,
+					Annotations: map[string]string{
+						rbacapi.AnnotationKeyManaged: rbacapi.AnnotationValueTrue,
+					},
+				},
+			},
+		); err != nil {
+			if kubeerr.IsAlreadyExists(err) {
+				saLogger.Debug("ServiceAccount already exists in project namespace")
+				continue
+			}
+			return fmt.Errorf(
+				"error creating ServiceAccount %q in project namespace %q: %w",
+				saName,
+				project.Name,
+				err,
+			)
+		}
+	}
+
+	roles := []*rbacv1.Role{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "admin",
+				Namespace: project.Name,
+				Annotations: map[string]string{
+					rbacapi.AnnotationKeyManaged: rbacapi.AnnotationValueTrue,
+				},
+			},
+			Rules: []rbacv1.PolicyRule{
```

Comment:
I slightly worry about the maintainability of this, as it's a separate place from where the other permissions are kept up-to-date, which already has a history of being forgotten at times.

Reply:
> as it's a separate place from where the other permissions are kept up-to-date

I think you're referring to the fact that the project webhook also manages some project-level permissions...

That wasn't ever intended to manage _all_ project-level permissions; rather it was intended to manage the very few for which there was a necessity of creating them synchronously. This is what enables you to define a Project at the top of a manifest and subsequently define resources that go in that Project's namespace. The webhook doesn't manage anything that isn't required to pull that off.

Reply:
Spoke offline and learned this is _not_ was @hiddeco was referring to. We're working it out offline.

Reply:
To recap what I meant: there is an overlap between the things defined here and the RBAC-related things that are defined in the chart. As we have a history of forgetting about updating permissions in the chart (#1907, #1798), I worry that this will be _another_ thing that is prone to bugs, as it is likely to go unnoticed during reviews.

However, both @krancour and I do not directly have a better solution for this, and we have agreed to revisit it in the future.